### ISSUE TITLE
Burn uses formatting language instead of UI language for localization

### ISSUE THREAD

**Issue by keithyipkw** [wixtoolset/issues#6120]
## Bug

* Which version of WiX are you building with?

3.11.2.4516

* Which version of Visual Studio are you building with (if any)?

2017 15.9.18

* Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?

1.0.0.4

* Which version of .NET are you building with?

I am not sure what this question is asking for.

* If the problem occurs when installing your packages built with WiX, what is the version of Windows the package is running on?

Windows 10 64bit Enterprise (Build 17763: Service Pack 0)

* Describe the problem and the steps to reproduce it.

Burn display texts in the language the same as the formatting language in Windows. The UI language in Windows has no effect. If I change the formatting language, the text in Burn changes to that language. In the logs, WixStdBALanguageId is the same as the formatting UI. UserUILanguageID is the same as the UI language in Windows.

My BootstrapperApplicationRef  is 

```
    <BootstrapperApplicationRef Id="WixStandardBootstrapperApplication.HyperlinkLicense">
      <bal:WixStandardBootstrapperApplication
        LicenseUrl=""
        SuppressOptionsUI="yes"
        ThemeFile="theme\HyperlinkTheme.xml"
        LocalizationFile="theme\HyperlinkTheme.wxl"
        xmlns:bal="http://schemas.microsoft.com/wix/BalExtension" />
      <Payload Id="theme-zh-cn" Compressed="yes" Name="2052\thm.wxl" SourceFile="theme\2052\HyperlinkTheme.wxl"/>
    </BootstrapperApplicationRef>
```

I am not sure if the fix cda30f74 in #5285 meant changing Burn to use UI language for localization.

* Describe the behavior you expected and how it differed from the actual behavior.

Burn uses UI "language" in Windows settings as the locale to display text.

**Comment by barnson** [comment]
That's what `UseUILanguages` does:

> When set to "yes", causes WixStdBA/Prereq BA to use the user's control panel  language settings. Otherwise, the previous API (which uses locale instead of  language) is used to maintain compatiblity with previous versions of WiX. On  Vista and newer platforms, this value set to "yes" will search all specified  user languages, including fallback languages, in order.  (http://schemas.microsoft.com/wix/BalExtension)

**Comment by keithyipkw** [comment]
> That's what `UseUILanguages` does:
> 
> > When set to "yes", causes WixStdBA/Prereq BA to use the user's control panel  language settings. Otherwise, the previous API (which uses locale instead of  language) is used to maintain compatiblity with previous versions of WiX. On  Vista and newer platforms, this value set to "yes" will search all specified  user languages, including fallback languages, in order.  (http://schemas.microsoft.com/wix/BalExtension)

Do you mean that using the formatted language is the expected behavior? Did the change in src/Setup/Bundle/WixStdBundle.wxs in https://github.com/wixtoolset/wix3/commit/cda30f74c8d77e2e2d6d514401de9588feab210d mean to use the UI language as the locale?

```
        <swid:Tag Regid="!(loc.Regid)" InstallPath="[ProgramFilesFolder]WiX Toolset v$(var.WixMajorMinor)" />
        <Update Location='!(loc.UpdateUrl)' />

-        <BootstrapperApplicationRef Id="WixStandardBootstrapperApplication.HyperlinkLicense">
+        <BootstrapperApplicationRef Id="WixStandardBootstrapperApplication.HyperlinkLicense" bal:UseUILanguages="yes">
            <bal:WixStandardBootstrapperApplication LicenseUrl="http://wixtoolset.org/about/license/" />
        </BootstrapperApplicationRef>
```

**Comment by BMurri** [comment]
**TL;DR** - in v3 always use the UseUILanguages  flag in all your bundles.

### Brief resumen of the gory details:

I believe that the originally expected behavior was to use the user's UI language, but please see #4499 if you wish to see more details.

The change you reference was intended to make the users of WiX's own installers get the expected behavior (UI language), and to model the use of the API. As a philosophy we err on the side of backwards compatibility and not breaking existing behavior as much as possible within any given version of the toolset. That's the reason #5285 was keep around as a v3-only "fix" to provide for the desired behavior without "silently" breaking someone else's working deployment.

We break those guarantees between major versions, so in v4 you always get UI language for localization.
